sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileExtensions.java

```diff
@@ -0,0 +1,23 @@
+package com.azure.storage.file;
+
+import com.azure.storage.common.Constants;
+import com.azure.storage.common.Utility;
+
+public class FileExtensions {
+
+    /**
+     * Verifies that the file permission and file permission key are not both set and if the file permission is set,
+     * the file permission is of valid length.
+     * @param filePermission The file permission.
+     * @param filePermissionKey The file permission key.
+     */
+    public static void filePermissionAndKeyHelper(String filePermission, String  filePermissionKey) {
```

Comment:
I think this should be removed and allow a request to be sent to the service and fail there. If the service changed to allow both values or a larger file permission size this would need to be updated or it would inappropriately fail requests.

Reply:
The permission size is required to be smaller than 8kb since headers cant be longer than that (request wont even go through) so I think it is best to keep that check. As for the other check, it is a conceptual check because it doesnt make sense to send both (it's always an either or).

Reply:
Agreed that it will fail to send. Generally we've erred on the side of less checks and allowing the service to handle invalid requests and send back an error message, that way if the service changes we don't need to do anything.